onnxruntime/core/graph/graph.cc

```diff
@@ -2055,6 +2056,39 @@ void Graph::RemoveInitializedTensor(const std::string& tensor_name) {
   }
 }
 
+Status Graph::ReplaceInitializedTensor(const ONNX_NAMESPACE::TensorProto& new_initializer) {
+  const auto& initializer_name = new_initializer.name();
+  auto name_to_initializer_it = name_to_initial_tensor_.find(initializer_name);
```

Comment:
>name_to_initial_tensor_ [](start = 32, length = 23)

I am not seeing that the map second is being updated.

Reply:
name_to_initial_tensor_ is a std::unordered_map<std::string, const ONNX_NAMESPACE::TensorProto*>, so we can't update the TensorProto through it. The actual TensorProto is in graph_proto_, which is modified below if all the checks pass.

Reply:
Should probably explain that in some comments here. It's somewhat indirect and obtuse, so it would greatly help anyone else looking at the code for that to be documented.

---
In reply to: [337778162](https://github.com/microsoft/onnxruntime/pull/2230#discussion_r337778162) [](ancestors = 337778162)